test/unit/orchestration/dhcp_test.rb

```diff
@@ -185,4 +185,15 @@ def teardown
     assert_equal 1, bmc.queue.items.select {|x| x.action == [ bmc,     :set_dhcp ] }.size
     assert_equal 1, bmc.queue.items.select {|x| x.action == [ bmc.old, :del_dhcp ] }.size
   end
+
+  test "new host with dhcp and no operating system should show correct validation on save" do
+    h = FactoryGirl.build(:host, :with_dhcp_orchestration)
+    h.operatingsystem = nil
```

Comment:
it could but the factory is built like this: 
``` 
architecture { operatingsystem.architectures.first }
medium { operatingsystem.media.first }
ptable { operatingsystem.ptables.first }
```
so I would have to use
```ruby
h = FactoryGirl.build(:host, :with_dhcp_orchestration, :operatingsystem => nil, :architecture => nil, :medium => nil, :ptable => nil)
```
and I though it was a little overkill for the test.